modules/templates/handler.rb

```diff
@@ -7,23 +7,25 @@ module Proxy::Templates
   class Handler < ::Proxy::HttpRequest::ForemanRequest
     extend Proxy::Log
 
-    def get_template(kind, token, static = false)
-      opts = { :token => token,
-               :url   => Proxy::Templates::Plugin.settings.template_url
+    def get_template(kind, params, static = false)
+      opts = {
+        :url => Proxy::Templates::Plugin.settings.template_url
       }
+      opts[:mac]    = params[:mac] if params.has_key?('mac')
```

Comment:
You are expecting string keys, but symbols are being sent instead (also the reason for test failures. Also, you can use ```rake test``` to execute tests locally).